tests/func/test_check_ignore.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+import pytest
+
+from dvc.ignore import DvcIgnore
+from dvc.main import main
+
+
+@pytest.mark.parametrize(
+    "file,ret,output", [("ignored", 0, "ignored\n"), ("not_ignored", 1, "")]
+)
+def test_check_ignore(tmp_dir, dvc, file, ret, output, caplog):
+    tmp_dir.gen(DvcIgnore.DVCIGNORE_FILE, "ignored")
+
+    assert main(["check-ignore", file]) == ret
+    assert output in caplog.text
+
+
+@pytest.mark.parametrize(
+    "file,ret,output",
+    [
+        ("file", 0, "{}:1:f*\tfile\n".format(DvcIgnore.DVCIGNORE_FILE)),
+        ("foo", 0, "{}:2:!foo\tfoo\n".format(DvcIgnore.DVCIGNORE_FILE)),
```

Comment:
This is what I would like to discuss.

Yes In Git, 0 should be returned when a given path is ignored. But 0 returns on any pattern match the file either included or excluded, 1 returns when none of the pattern match. (It might be a bug or a feature not clearly documented. I sent an email to Git last time but no reply)
But I felt illogical on this results. 0 for ignored and 1 for included or not matched might be more reasonable. 

As a result. I left `matches` in
```
CheckIgnoreResult = namedtuple(
    "CheckIgnoreResult", ["file", "matches", "patterns"]
)
``` 
which could be used for marking whether it is ignored or not.

Reply:
@karajan1001 
To me, it seems that either docs is wrong (because it says that 0 is returned if file is ignored) or implementation (which returns 0 upon any matching spec). I'll try to reach git-devel IRC, maybe we can get an answer faster. 
For now, we can leave it as is.